instrumentation/opentelemetry-instrumentation-starlette/src/opentelemetry/instrumentation/starlette/__init__.py

```diff
@@ -169,8 +169,11 @@ def client_response_hook(span: Span, scope: dict[str, Any], message: dict[str, A
 API
 ---
 """
+# pyright: reportPrivateUsage=false
```

Comment:
I use VSCode, and this would simplify me contributing to this project.

Reply:
What does it not like? If it's in some common pattern we can make it project wide maybe?

Reply:
The `reportPrivateUsage` is for attribution on private attribute, e.g.:

```py
        _InstrumentedStarlette._tracer_provider = kwargs.get("tracer_provider")
```
`_tracer_provider` is private.

---

> If it's some common pattern shared by other packages we can make it project wide maybe?

All VSCode users already have [pyright](https://github.com/microsoft/pyright). But then it's a bigger discussion... This project already has a non-used (?) mypy dependency (old version btw) on `dev-requirements.txt`. A project should only use one of them... Both doesn't make sense.

Reply:
If this is blocker for this PR, happy to remove it.

I use my IDE in strict mode... So...

Reply:
As far as I understand, this can be a problem only if we start running Pyright at CI for instrumentation packages, right?